nanowar.ts

```diff
@@ -0,0 +1,314 @@
+            state.tick.troops.push({
+                id: troopIDCounter++,
+                from: troop.from,
+                to: troop.to,
+                player: troop.playerID,
+                size: troop.size,
+                endTick: state.tick.id + state.planetsDistances[troop.from][troop.to],
+            });
+        }
+    }
+
+    // Process troops that are arriving to their planets, preparing for fight
+    let planetWaitingList: {who: PlayerID, size: number}[][] = Array.from(Array(state.planets.length), () => []); // TODO: use dictionary instead of array
+    for (let i = 0; i < state.tick.troops.length; i++) {
+        if (state.tick.troops[i].endTick === state.tick.id) {
+            let planet = state.tick.troops[i].to;
+            let user = { who: state.tick.troops[i].player, size: state.tick.troops[i].size };
+            // Add planets to
+            planetWaitingList[planet].push(user);
+            console.log("Arrived:", planetWaitingList);
+            state.tick.troops.splice(i, 1); // Removing troop from list
+            i--;
+        }
+    }
+    // FIGHT!
+    for (let i=0; i < planetWaitingList.length; i++) { // i = planetID
+        // No one is coming to this planet
+        let planet = planetWaitingList[i];
+        if (planet.length === 0) {
+            continue;
+        }
+        // Preparing some variables
+        let sizes = new Array(state.players.length+1).fill(0); // TODO: use dictionary instead of array
+        for (let user of planet) {
+            sizes[user.who] += user.size;
+        }
+        // Add neutral planet population to the fight
+        sizes[state.players.length] = state.tick.planets[i].player === null ? state.tick.planets[i].population : 0;
+        // Planet owner is coming to this planet
+        let planetOwner = state.tick.planets[i].player;
+        if (planetOwner !== null) { // TODO: state.tick.planets[i].player === null
+            sizes[planetOwner.id] += state.tick.planets[i].population;
+        }
+        //console.log("Planet:", state.tick.planets[i]);
+        //console.log("Sizes:", sizes);
+        // Get the two biggest sizes
+        let max : {who: number | null, size: number} = {who: null, size: 0};
+        let max2 : {who: number | null, size: number} = {who: null, size: 0};
+        for (let i = 0; i < state.players.length+1; i++) {
+            if(sizes[i] > max.size){
+                max2 = max;
+                max = {who: i, size: sizes[i]};
+            } else if (sizes[i] > max2.size) {
+                max2 = {who: i, size: sizes[i]};
+            }
+        }
+        //console.log("Maxok:", max, max2)
+        // Determine the winner, update the planet
+        if (max.size === max2.size){
+            state.tick.planets[i].player = null;
+            state.tick.planets[i].population = 0;
+        } else if (max.who !== null) {
+            state.tick.planets[i].player = {id: max.who, startingTick: state.tick.id};
```

Comment:
Ha eddig is az övé volt, akkor jogos, hogy reseteljük a startingTick-et?

Reply:
Jogos, javítottam.

Reply:
Ez amúgy egy nagyon fontos észrevétel volt, mert a javítás előtt meg lehetett azt csinálni, hogy folyamatosan küldesz 1-1 troopot az ellenfél bolygójára, így ő nem kap soha sem bónuszt, mert a startingTick folyamatosan frissül.